Society for Neuroscience Abstract 2016.js

```diff
@@ -0,0 +1,446 @@
+		//Zotero.debug(items);
+
+		Zotero.selectItems(items, function(selectedItems){
+
+		if(!selectedItems) return true;
+
+			var abstractIDs = [];
+			for (var i in selectedItems) {
+				abstractIDs.push(i);
+				//ZU.processDocuments(i, scrape);
+			}
+
+			var baseurl = url.replace(/presentations.+$|sessions.+$|participants.+$/,"presntation/");
+			Zotero.debug(baseurl);
+			scrape(abstractIDs,baseurl);
+		});
+
+	}
+}
+
+
+
+
+function scrape(ids,baseurl){
+
+Zotero.debug(ids);
+
+for (k = 0; k < ids.length; k++) {
+
+	var newurl = baseurl + ids[k];
+	Zotero.debug(newurl);
+
+	ZU.processDocuments(newurl, function (doc) {
+		// NOTE doGet()  html does not contain the actual contents,
+		// object is HTTPreqest object, use processDocuments
+		//
+		// Zotero.debug(doc); //object HTMLDocument
+		// Zotero.debug(doc.URL);
+
+		var item = new Zotero.Item("conferencePaper");
+
+		// pages, title
+		var color_primary = ZU.xpath(doc,'//h1[@class="color-primary"]');
+
+		//TODO when multiple doc does not contain the contents
+
+		// Zotero.debug(color_primary);
+		//TODO does not work for mulitiple
+
+		// var m1 = color_primary[0].innerHTML.match(/^\s*(\d+\.\d+)\n\s*\/\s(\w+\d+)\n\s*\-\s/);
+
+		// 282.18/B8, 1, 3.02
+		var m1 = color_primary[0].innerHTML.match(
+			/^\s*\n\s*(\d+)(\.(\d+))?\n(\s*\/\s([\w\/\d]+)\n)?\s*-\s/);
+
+		//Zotero.debug(m1);
+
+		var title;
+		if (m1 != null ) {
+			if (m1[3] != undefined) {
+				if (m1[5] != undefined) {
+					item.pages = m1[1] + "." + m1[3] + "/" + m1[5];
+				} else {
+					item.pages = m1[1] + "." + m1[3];
+				}
+			} else {
+				item.pages = m1[1];
+			}
+
+			title = color_primary[0].innerHTML.replace(
+				/^\s*\n\s*\d+(\.\d+)?\n(\s*\/\s[\w\/\d]+\n)?\s*-\s/,"");
+		} else {
+			// SPC01, SOC01, SAT02, PDW01
+
+			var m2 = color_primary[0].innerHTML.match(/^[\s\n]*([\w\d]+)\n.+\s*-\s/);
+			//Zotero.debug(m2);
+
+			item.pages = m2[1];
+
+			var m3 = color_primary[0].innerHTML.match(/\n\s+-\s(.+)\t+<br>\t/);
+			if (m3 != null) {
+				title = m3[1];
+			} else {
+				title = "";
+			}
+			//Zotero.debug(m3[1]); // null
+
+		}
+		if (title == title.toUpperCase()) {
+			title = ZU.capitalizeTitle(title,true);
+		};
+
+		item.title = title;
+
+		//Zotero.debug(item.pages);
+		//Zotero.debug(item.title);
+
+
+		// extra
+		var well =ZU.xpath(doc,'//div[@class="well well-small"]');
+		//Zotero.debug(well[0].innerHTML);
+
+		var mst = well[0].innerHTML.match(/<dt>Session Type<\/dt>[.\n\s]*<dd>(.+)<\/dd>/);
+
+		if (mst == null) {
+			mst = "";
+		}
+
+		//Zotero.debug(mst);
+		var session = ZU.xpath(doc,'//h2[@class="session"]/a');
+
+		item.extra = "Session: " + session[0].innerText
+			+ "; " + "Session Type: " + mst[1];
+
+
+		//attachments
+		//TODO does not work
+		// item.attachments = [{
+		//	url: doc.URL,
+		//	title: "Print page",
+		//	mimeType: "text/html",
+		//	snapshot: true
+		//}];
+
+
+		// creators
+
+		var disclosure = ZU.xpath(doc,'//div[@class="span7"]/dl/dd[2]'); // Use Disclosure data
```

Comment:
Extracting author names was one of the trickiest bits.

Initially, I tried to take out from **Authors** field of the page content. But names in this were all in upper case, with annoying `<sup>1</sup>` for affiliation etc. Instead, names in **Disclosure** field are starting with uppercase followed by the lower case letters in a normal way.

It's shame they don't use `id` attribute for such important elements.

Reply:
Okay, I see. I don't like to take the second `dd` element which can change quite frequently. Usually I prefer to have something semantically meaningful or more stable.

Reply:
Here is an alternative code which might even work for the case where there is no disclosure statement:
```js
		var disclosures = ZU.xpath(doc, '//dl/dt[contains(., "Disclosures")]/following-sibling::dd');
		var authorsList = ZU.xpath(doc, '//dl/dt[contains(., "Authors")]/following-sibling::dd');
		if (disclosures) {
			var authors = ZU.xpath(disclosures[0], './b');
		} else if (authorsList) {
			var authors = authorsList.split(',');
		}
		if (authors) {
			for (var i=0; i<authors.length; i++) {
				item.creators.push(ZU.cleanAuthor((authors[i].textContent), "author"));
			}
		}
```
You might need to further optimize this. I haven't tested this code much.

Reply:
Thanks, it worked with a few examples and the xpath specification seems more robust.

The reason I chose Disclosures was that Authors were all upper case in the case of Society for Neuroscience. But this may not hold true for other conferences.

Reply:
Yeah, I have seen the all-uppercases names. For me it is fine to concentrate on the cases you are specially interested in and try to do something reasonable for other possible cases.